server/mqtt.go

```diff
@@ -3823,7 +3829,7 @@ CHECK:
 			// This Session lasts as long as the Network Connection. State data
 			// associated with this Session MUST NOT be reused in any subsequent
 			// Session.
-			if err := es.clear(); err != nil {
+			if err := es.clear(false); err != nil {
```

Comment:
At least here we make sure that the delete is complete, which I think is the right thing to do.

Reply:
Right. I understand why checking the success is important here, thus preserved as before.